### DISCUSSION TITLE
[Intel GPU] how to know the SIMD width compiled from dpc++ code

### DISCUSSION THREAD

**Discussion by guoyejun** [intel/llvm#4291] [Q&A] [answered]
I'm working on Intel GPU with:
```
queue q(gpu_selector{}, propList);
```

And my dpcpp code is normal, looks like:
```
event add_data(queue q, float *input0, float *input1, float *output)
{
    auto event = q.submit([&](sycl::handler &h) {
        h.parallel_for<class ker_1>(len, [=](cl::sycl::item<1> item) {
            int idx = item.get_id(0);
            output[idx] = input0[idx] + input1[idx];
        });
    });
    return event;
}
```

I just use the simple way to compile the code:
```
dpcpp mycode.cpp
```

My question is, what SIMD width will be used, for example, SIMD-8, SIMD-16 or SIMD-32 for the GPU instruction? How could I query it? thanks.

**Comment by bader** [comment, marked as answer]
> My question is, what SIMD width will be used, for example, SIMD-8, SIMD-16 or SIMD-32 for the GPU instruction? How could I query it?

@PawelJurek, which API exposes this kernel property?

**Reply by PawelJurek** [reply]
@bader , please correct me if I'm wrong, but it seems that this value can be queried via [SYCL_INTEL_sub_group extension](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc)

**Reply by bader** [reply]
You seem to be right. Thanks for pointing this out!

**Comment by guoyejun** [comment]
thanks, got it.

btw, is it possible to dump the GPU asm, and so we can also know the SIMD width from the kernel asm code.

**Reply by PawelJurek** [reply]
Yes, this is available on Linux drivers via IGC_ShaderDumpEnable flag. Instructions are here: https://github.com/intel/intel-graphics-compiler/blob/master/documentation/configuration_flags.md

**Reply by guoyejun** [reply]
thanks!

**Comment by guoyejun** [comment]
not know how to change the status to be answered

**Reply by bader** [reply]
This should help - https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer
